sklearn/tests/test_base.py

```diff
@@ -508,10 +508,8 @@ def fit(self, X, y=None):
             return self
 
     est = MyEstimator()
-    with pytest.warns(FutureWarning, match='AttributeError'):
-        params = est.get_params()
 
-    assert params['param'] is None
+    assert_raises(AttributeError, est.get_params)
```

Comment:
We can use `with pytest.raises(AttributeError, match=msg)` here as well.

Reply:
I decide to use the validation tools of scikit-learn, but I have no preference. If `pytest.raises` is preferred over `assert_raises` (or `assert_raises_regex`), I will change it.

Reply:
It is preferred.

The only place where we prefer `assert_raises` is in `test_estimator_checks` which assume we do not have pytest installed.

Reply:
Good point. I will keep this in mind for future reference!

Reply:
@thomasjpfan Should we mention that in `test_estimator_checks`?